client-spark/spark-2/src/main/scala/org/apache/spark/shuffle/celeborn/CelebornShuffleFallbackPolicyRunner.scala

```diff
@@ -19,25 +19,32 @@ package org.apache.spark.shuffle.celeborn
 
 import org.apache.celeborn.client.LifecycleManager
 import org.apache.celeborn.common.CelebornConf
+import org.apache.celeborn.common.exception.CelebornIOException
 import org.apache.celeborn.common.internal.Logging
+import org.apache.celeborn.common.protocol.FallbackPolicy
 
 class CelebornShuffleFallbackPolicyRunner(conf: CelebornConf) extends Logging {
 
   def applyAllFallbackPolicy(lifecycleManager: LifecycleManager, numPartitions: Int): Boolean = {
-    applyForceFallbackPolicy() || applyShufflePartitionsFallbackPolicy(numPartitions) ||
-    !checkQuota(lifecycleManager) || !checkWorkersAvailable(lifecycleManager)
+    val needFallback =
+      applyForceFallbackPolicy() || applyShufflePartitionsFallbackPolicy(numPartitions) ||
+        !checkQuota(lifecycleManager) || !checkWorkersAvailable(lifecycleManager)
+    if (needFallback && FallbackPolicy.NEVER.equals(conf.shuffleFallbackPolicy)) {
+      throw new CelebornIOException("Fallback to Spark's default shuffle is prohibited.")
+    }
+    needFallback
```

Comment:
Only a nit, thanks. Maybe we can save shuffleFallbackPolicy as a member variable and clear the judge logic like:
```
    if (FallbackPolicy.ALWAYS.equals(shuffleFallbackPolicy)) {
      logWarning(..)
      true
    } else {
       val needFallback = ..
       if (needFallback && FallbackPolicy.NEVER.equals(shuffleFallbackPolicy)) {
          throw .. 
       }
       needFallback
    }
```

Reply:
@littlexyw, could you take above comment?